OkvisSLAMSystem.cpp

```diff
@@ -151,29 +154,50 @@ namespace ark {
                     keyframe->descriptors_[cam_idx]=frame_data.data->descriptors[cam_idx];
                 }
 
-                for (int i = 0; i < sparse_map_vector.size()-1; i++) {
-                    const auto sparseMap = sparse_map_vector[i];
-                    if (sparseMap->detectLoopClosure(keyframe)) {
-                        active_map_index = i;
-                        // delete all the new map after active map
-                        sparse_map_vector.resize(active_map_index+1);
-                        cout << "Deleting newer maps after: " << active_map_index << endl;
-                        for (MapSparseMapDeletionHandler::const_iterator callback_iter = mMapSparseMapDeletionHandler.begin();
-                            callback_iter != mMapSparseMapDeletionHandler.end(); ++callback_iter) {
-                            const MapSparseMapDeletionHandler::value_type& pair = *callback_iter;
-                            pair.second(i);
+                // apply correction
+                keyframe->T_WS_ = correction_ * keyframe->T_WS_;
+                MapKeyFrame::Ptr loop_kf = nullptr;
+                Eigen::Affine3d transformEstimate;
+                if (detectLoopClosure(keyframe, loop_kf, transformEstimate)) {
+                    //cout << "Loop closure detected" << endl;
+                    addKeyFrameResult = true;
+                    for (int i = 0; i < sparse_map_vector.size()-1; i++) {
```

Comment:
This for loop is concerning. Why are we not iterating to the current (active) SparseMap? There should be loop closure functionality within the current active map along with loop closures to inactive maps. I checked back to the previous version of this code that Yohan's team pushed last semester and it seemed to have this same issue, that the loop closures were only checked against inactive maps. Is there any functionality to do loop closures with the current map (in which case, mergeMaps wouldn't be called)?

Reply:
seems right, addkeyframe resolves current map loop closure